enumerables.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      if block_given?
+        yield(to_a[i])
+        i += 1
+      end
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      return to_enum if block_given?
```

Comment:
- [ ] my_each_with_index calls the given block once for each element in self

```
irb(main):147:0> [1,2,3].each_with_index do |i,ind| puts i,ind end
1
0
2
1
3
2
=> [1, 2, 3]
irb(main):148:0> [1,2,3].my_each_with_index do |i,ind| puts i,ind end
=> #<Enumerator: [1, 2, 3]:each>
```

- [ ] enumerables my_each_with_index returns an enumerator if no block is given